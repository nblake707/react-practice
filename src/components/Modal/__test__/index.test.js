import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Modal from '../../Modal';

// need to mock props (toggle function) passed to the modal component. 
const mockToggleModal = jest.fn(); 

// set up to mimic currentPhoto prop
const currentPhoto = {
    name: 'park bench',
    category: 'landscape',
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie',
    index: 1
};

afterEach(cleanup);

describe('Modal Component', () => {
    it('renders', () => {
        render(
            <Modal
              currentPhoto={currentPhoto}
              onClose={mockToggleModal}
            />
        )
    });

    it("renders", () => {
      const { asFragment } = render(
        <Modal currentPhoto={currentPhoto} onClose={mockToggleModal} />
      );
      expect(asFragment()).toMatchSnapshot();
    });
});

describe('Click Event', () => {
  it('calls onClose handler', () => {
    const { getByText } = render(
      <Modal 
        onClose={mockToggleModal}
        currentPhoto={currentPhoto}
      />
    )
    fireEvent.click(getByText('close this modal'));
    expect(mockToggleModal).toHaveBeenCalledTimes(1);
  })
})



