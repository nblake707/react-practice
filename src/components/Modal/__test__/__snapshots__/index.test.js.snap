// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component renders 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
      >
        park bench
      </h3>
      <img
        alt="current category"
      />
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie
      </p>
      <button
        type="button"
      >
        close this modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;
